frame/staking/src/pallet/impls.rs

```diff
@@ -1499,59 +1485,68 @@ impl<T: Config> SortedListProvider<T::AccountId> for UseNominatorsAndValidatorsM
 			Err(())
 		}
 	}
+
 	fn count() -> u32 {
 		Nominators::<T>::count().saturating_add(Validators::<T>::count())
 	}
+
 	fn contains(id: &T::AccountId) -> bool {
 		Nominators::<T>::contains_key(id) || Validators::<T>::contains_key(id)
 	}
+
+	fn get_score(id: &T::AccountId) -> Result<Self::Score, Self::Error> {
+		Ok(Pallet::<T>::weight_of(id))
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn try_state() -> Result<(), &'static str> {
+		Ok(())
+	}
+
 	fn on_insert(_: T::AccountId, _weight: Self::Score) -> Result<(), Self::Error> {
 		// nothing to do on insert.
 		Ok(())
 	}
-	fn get_score(id: &T::AccountId) -> Result<Self::Score, Self::Error> {
-		Ok(Pallet::<T>::weight_of(id))
-	}
+
 	fn on_update(_: &T::AccountId, _weight: Self::Score) -> Result<(), Self::Error> {
 		// nothing to do on update.
 		Ok(())
 	}
+
 	fn on_remove(_: &T::AccountId) -> Result<(), Self::Error> {
 		// nothing to do on remove.
 		Ok(())
 	}
+
+	frame_election_provider_support::runtime_benchmarks_or_test_enabled! {
+		fn unsafe_clear() {
+			// NOTE: Caller must ensure this doesn't lead to too many storage accesses. This is a
+			// condition of SortedListProvider::unsafe_clear.
+			#[allow(deprecated)]
+			Nominators::<T>::remove_all();
+			#[allow(deprecated)]
+			Validators::<T>::remove_all();
+		}
+
+		fn score_update_worst_case(_who: &T::AccountId, _is_increase: bool) -> Self::Score {
+			unimplemented!()
+		}
+	}
+
 	fn unsafe_regenerate(
 		_: impl IntoIterator<Item = T::AccountId>,
 		_: Box<dyn Fn(&T::AccountId) -> Self::Score>,
 	) -> u32 {
 		// nothing to do upon regenerate.
 		0
 	}
-
-	#[cfg(feature = "try-runtime")]
-	fn try_state() -> Result<(), &'static str> {
-		Ok(())
-	}
-
-	fn unsafe_clear() {
-		// NOTE: Caller must ensure this doesn't lead to too many storage accesses. This is a
-		// condition of SortedListProvider::unsafe_clear.
-		#[allow(deprecated)]
-		Nominators::<T>::remove_all();
-		#[allow(deprecated)]
-		Validators::<T>::remove_all();
-	}
-
-	#[cfg(feature = "runtime-benchmarks")]
-	fn score_update_worst_case(_who: &T::AccountId, _is_increase: bool) -> Self::Score {
-		unimplemented!()
-	}
 }
 
 // NOTE: in this entire impl block, the assumption is that `who` is a stash account.
 impl<T: Config> StakingInterface for Pallet<T> {
-	type AccountId = T::AccountId;
 	type Balance = BalanceOf<T>;
```

Comment:
TODO for me: re-review once diff is updated.

Reply:
re-ordering this stuff makes the PR unnecessarily diccicult to review, would be good if you revert those.

Reply:
It's reordered according to the actual trait definition to match it. But yeah, I have already realized it might be painful to review.

Reply:
fixed